cap/modules/mail/receivers.py

```diff
@@ -21,40 +21,65 @@
 # In applying this license, CERN does not
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
-from flask import request
+from flask import current_app
 
-from .utils import NOTIFICATION_RECEPIENT, generate_notification_attrs, \
-    send_mail_on_review, send_mail_on_publish, create_analysis_url
+from .attributes import generate_recipients, generate_message,\
+    generate_subject, generate_template
+from .custom.messages import working_url
+from .custom.subjects import draft_id, published_id, revision
+from .tasks import create_and_send
 
 
 def post_action_notifications(sender, action=None, pid=None, deposit=None):
-    """Method to run after a deposit action ."""
-    schema = deposit.get("$schema")
-    recipients_config = NOTIFICATION_RECEPIENT.get(schema, {}).get(action)
-    host_url = request.host_url
-
-    if recipients_config:
-        subject, message, recipients = generate_notification_attrs(
-            deposit, host_url, recipients_config)
-
-        if recipients:
-            if action == "publish":
-                recid, record = deposit.fetch_published()
-
-                send_mail_on_publish(
-                    recid.pid_value,
-                    record.revision_id,
-                    host_url,
-                    recipients,
-                    message,
-                    subject_prefix=subject)
-
-            if action == "review":
-                analysis_url = create_analysis_url(deposit)
-
-                send_mail_on_review(
-                    analysis_url,
-                    host_url,
-                    recipients,
-                    message,
-                    subject_prefix=subject)
+    """
+    Notification through mail, after specified deposit actions.
+    The procedure followed to get the mail attrs will be described here:
+
+    - Get the config for the action that triggered the receiver.
+    - Through the configuration, retrieve the recipients, subject, template
+      and message, and render them when needed.
+    - Create the message and mail contexts (attributes), and pass them to
+      the `create_and_send` task.
+    """
+    mail_sender = current_app.config.get('MAIL_DEFAULT_SENDER')
+    if not mail_sender:
+        current_app.logger.error('Mail Error: Sender not found.')
+
+    action_configs = deposit.schema.config.get('notifications', {}) \
+        .get('actions', {}) \
+        .get(action, [])
+
+    for config in action_configs:
+        recipients, cc, bcc = generate_recipients(deposit, config)
+
+        if not any([recipients, cc, bcc]):
+            current_app.logger.error(
+                f'Mail Error for analysis with id {deposit["_deposit"]["id"]}:\n'  # noqa
+                f'Empty recipient list.')
+            return
```

Comment:
if one of the configs fails he (e.g. no recipients, bcc, cc), it will not move on to other configs. No?